flink-python/pyflink/table/table.py

```diff
@@ -96,16 +105,413 @@ def where(self, predicate):
+
+        Example:
+
+        Returns the first 3 records.
+        ::
+            >>> tab.order_by("name.desc").fetch(3)
+
+        Skips the first 10 rows and returns the next 5 rows.
+        ::
+            >>> tab.order_by("name.desc").offset(10).fetch(5)
+
+        :param fetch: The number of records to return. Fetch must be >= 0.
+        :return: Result table.
+        """
+        return Table(self._j_table.fetch(fetch))
+
+    def window(self, window):
+        """
+        Defines group window on the records of a table.
+
+        A group window groups the records of a table by assigning them to windows defined by a time
+        or row interval.
+
+        For streaming tables of infinite size, grouping into windows is required to define finite
+        groups on which group-based aggregates can be computed.
+
+        For batch tables of finite size, windowing essentially provides shortcuts for time-based
+        groupBy.
+
+        .. note::
+            Computing windowed aggregates on a streaming table is only a parallel operation
+            if additional grouping attributes are added to the
+            :func:`~pyflink.table.GroupWindowedTable.group_by` clause.
+            If the :func:`~pyflink.table.GroupWindowedTable.group_by` only references a GroupWindow
+            alias, the streamed table will be processed by a single task, i.e., with parallelism 1.
+
+        :param windows: A :class:`GroupWindow` created from :class:`Tumble`, :class:`Session` or
+                        :class:`Slide`.
+        :return: A :class:`GroupWindowedTable`.
+        """
+        # type: (GroupWindow) -> GroupWindowedTable
+        return GroupWindowedTable(self._j_table.window(window._java_window))
+
+    def over_window(self, *over_windows):
+        """
+        Defines over-windows on the records of a table.
+
+        An over-window defines for each record an interval of records over which aggregation
+        functions can be computed.
+
+        Example:
+        ::
+            >>> table.window(Over.partition_by("c").order_by("rowTime")\
+            ...     .preceding("10.seconds").alias("ow"))\
+            ...     .select("c, b.count over ow, e.sum over ow")
+
+        .. note::
+            Computing over window aggregates on a streaming table is only a parallel
+            operation if the window is partitioned. Otherwise, the whole stream will be processed
+            by a single task, i.e., with parallelism 1.
+
+        .. note::
+            Over-windows for batch tables are currently not supported.
+
+        :param windows: :class:`OverWindow`s created from :class:`Over`.
+        :return: A :class:`OverWindowedTable`.
+        """
+        gateway = get_gateway()
+        window_array = gateway.new_array(gateway.jvm.OverWindow, len(over_windows))
+        for i in xrange(0, len(over_windows)):
+            window_array[i] = over_windows[i]._java_over_window
```

Comment:
can make use of the utility method `to_jarray`

Reply:
Done.